src/main/java/seedu/address/logic/parser/CsvAssignmentParser.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.logic.parser;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.assignment.Assignment;
+
+/**
+ * Parses assignment data from a CSV field, with each assignment represented as "assignmentName | score".
+ * The parser validates assignment names and scores and maps them to Assignment objects.
+ */
+public class CsvAssignmentParser {
+
+    /**
+     * Parses the assignments field, where each assignment is represented as "assignmentName | score".
+     *
+     * @param assignmentField string representing assignments and their scores
+     * @param model           the application model
+     * @return a map of assignment names to Assignment objects
+     * @throws CommandException if assignment name or score is invalid
+     */
+    public static Map<String, Assignment> parseAssignment(String assignmentField, Model model)
+            throws NumberFormatException, CommandException {
+        Map<String, Assignment> assignments = new HashMap<>();
+        assignmentField = assignmentField.trim();
+
+        if (!assignmentField.isEmpty()) {
+            String[] assignmentArray = assignmentField.split(",");
+            for (String assignment : assignmentArray) {
+                processAssignment(assignment.trim(), assignments, model);
+            }
+        }
+        return assignments;
+    }
+
+    /**
+     * Processes each assignment entry by validating the name and score, and adds it to the assignments map.
+     *
+     * @param assignment  string representing a single assignment
+     * @param assignments map to store validated assignments
+     * @param model       the application model
+     * @throws CommandException if assignment name or score is invalid
+     */
+    public static void processAssignment(String assignment, Map<String, Assignment> assignments, Model model)
```

Comment:
I don't think this method was used anywhere outside of this class, might be better to make this private.